mapping.Rmd

```diff
@@ -31,1098 +33,1520 @@ options(scipen = 999)
+```{r geofacet-data}
+library(geofacet)
 
-Dot maps are often more informative and convincing than choropleth maps, but data with geographic variables are usually defined by geography level (Census tract, county, etc.) instead of longitude and latitude. 
+head(state_ranks %>% as_tibble())
+```
 
-If the data are grouped by small geographies, like a Census tracts, coordinates can be randomly sampled from the shapefile for that geography. In this case, the information gained by being able to see the geographic distribution is often justifies the small amount of sampling error. 
+```{r geofacet-ex, cache = TRUE}
+set_urbn_defaults(style = "print")
 
-The Housing Finance Policy Center's ["An interactive view of the housing boom and bust"](http://apps.urban.org/features/mortgages-by-race/) uses data with sampled longitudes and latitudes from Census track shapefiles for millions of mortgages across sixteen years. The following example is based on the code from that project. 
+state_ranks %>%
+  filter(variable %in% c("education", "employment")) %>%
+  ggplot(aes(x = rank, y = variable)) +
+  geom_col() +
+  facet_geo(
+    facets = "state",
+    # Use custom urban geofacet grid which is built into urbnthemes
+    # For now we need to rename a few columns as urbnthemes has to be
+    # updated
+    grid = urbnthemes::urbn_geofacet %>%
+      rename(
+        code = state_code,
+        name = state_name
+      )
+  )
+```
 
-The two biggest challenges in sampling coordinates are usually: 
-1) Finding the correct shapefile because geographies change over time and geographic features like water exist in the middle of geographies. 
-2) Using and merging the shapefile in R because shapefiles are usually complicated hierarchical data structures. 
+Interactive geofacets of the United States have been used in Urban
+Features like [A Matter of
+Time](https://apps.urban.org/features/long-prison-terms/trends.html)
+which included geofaceted line charts showing trends in incarceration by
+state. Static geofacets of the United States were included in [Barriers
+to Accessing Homeownership Down Payment, Credit, and
+Affordability](https://www.urban.org/sites/default/files/publication/94801/barriers-to-homeownership-down-payments-credit-access-and-affordability_3.pdf)
+by the Housing Finance Policy Center.
+
+## Cartograms
+
+Cartograms are a modified form of a choropleth map with intentionally
+distorted sizes that map to a variable in your data. Below we create a
+cartogram with `library(cartogram)` where the state sizes are
+proportional to the population.
+
+```{r cartogram-example, cache = TRUE}
+library(cartogram)
+
+set_urbn_defaults(style = "print")
+
+chip_with_geographies_weighted <- chip_with_geographies %>%
+  # Note column name needs to be in quotes for this package
+  cartogram_cont(weight = "population")
+
+ggplot() +
+  geom_sf(
+    data = chip_with_geographies_weighted,
+    # Color in states by chip percentages
+    aes(fill = chip_pct)
+  )
```

Comment:
This legend is sloppy

Reply:
Yup that's bc I accidentally set urban style to print, instead of map. Now fixed!